app/server/appsmith-git/src/main/java/com/appsmith/git/service/GitExecutorImpl.java

```diff
@@ -90,4 +98,36 @@ private Path createRepoPath(String organizationId, String defaultApplicationId)
         return Paths.get(fileUtils.getGitRootPath(), organizationId, defaultApplicationId);
     }
 
+    @Override
+    public String cloneApp(String repoPath,
+                           String repoName,
+                           String remoteUrl,
+                           String privateSshKey,
+                           String publicSshKey) throws GitAPIException, IOException {
+        File file = getFilePath(repoPath, repoName);
+        final TransportConfigCallback transportConfigCallback = new SshTransportConfigCallback(privateSshKey, publicSshKey);
+        Git result = Git.cloneRepository()
+                .setURI(remoteUrl)
+                .setTransportConfigCallback(transportConfigCallback)
+                .setDirectory(file)
+                .call();
+        return result.getRepository().getBranch();
+    }
+
+    /* There might be a case where the name conflicts can occur while creating the file.
```

Comment:
Nit: 
```suggestion
    /** There might be a case where the name conflicts can occur while creating the file.
```

Adding the 2 `**` converts the comment to a function doc instead of a normal comment.